libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,15 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	//if a block has been added via mining or other block import function
+	//through RPC, then we should count it as a last imported block
+	RecursiveGuard l(x_sync);
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
`m_lastImportedBlock` is the last block pushed to `BlockQueue` and during sync `BlockQueue` has block numbers further ahead of those currently being imported in `BlockChain`